src/sortStudents.ts

```diff
@@ -1,16 +1,73 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[]
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(student: Student): number {
+  const { grades } = student;
+  const gradesLength = grades.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  if (gradesLength === 0) {
+    return 0;
+  }
+
+  return grades.reduce((total, grade) => total + grade) / gradesLength;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return sortedStudents.sort((s1, s2) => {
+        return order === 'asc'
+          ? s1.name.localeCompare(s2.name)
+          : s2.name.localeCompare(s1.name);
+      });
+    case SortType.Surname:
+      return sortedStudents.sort((s1, s2) => {
+        return order === 'asc'
+          ? s1.surname.localeCompare(s2.surname)
+          : s2.surname.localeCompare(s1.surname);
+      });
```

Comment:
I think you could combine this two cases like:

```suggestion
    case SortType.Name:
    case SortType.Surname:
      return sortedStudents.sort((s1, s2) => {
        return order === 'asc'
          ? s1[sortBy].localeCompare(s2[sortBy])
          : s2[sortBy].localeCompare(s1[sortBy]);
      });
```

Reply:
I merged these cases.